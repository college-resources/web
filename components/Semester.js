import React from 'react'
import Typography from '@material-ui/core/Typography'
import { withStyles, makeStyles } from '@material-ui/core/styles'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import MuiExpansionPanel from '@material-ui/core/ExpansionPanel'
import MuiExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary'
import MuiExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails'
import { lightBlue } from '@material-ui/core/colors'

const ExpansionPanelDetails = withStyles(theme => ({
  root: {
    padding: theme.spacing(2)
  }
}))(MuiExpansionPanelDetails)

const StyledTableCell = withStyles(theme => ({
  head: {
    backgroundColor: theme.palette.secondary.main,
    color: theme.palette.common.white
  },
  body: {
    fontSize: 14
  }
}))(TableCell)

const StyledTableRow = withStyles(() => ({
  root: {
    '&:nth-of-type(even)': {
      backgroundColor: lightBlue[50]
    },
    '&:nth-of-type(odd)': {
      backgroundColor: lightBlue[100]
    }
  }
}))(TableRow)

const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing(1),
    overflowX: 'auto'
  }
}))

const ExpansionPanel = withStyles({
  root: {
    border: 'none',
    boxShadow: 'none',
    '&:not(:last-child)': {
      borderBottom: 0
    },
    '&:before': {
      display: 'none'
    },
    '&$expanded': {
      margin: 'auto'
    }
  },
  expanded: {}
})(MuiExpansionPanel)

const ExpansionPanelSummary = withStyles(theme => ({
  root: {
    backgroundColor: theme.palette.secondary.dark,
    color: '#fff',
    marginBottom: -1,
    minHeight: 56,
    '&$expanded': {
      minHeight: 56
    }
  },
  expanded: {}
}))(MuiExpansionPanelSummary)

export default function Semester (props) {
  const classes = useStyles()
  const rows = props.rows
  const [expanded, setExpanded] = React.useState('panel1')
  const handleChange = panel => (event, newExpanded) => {
    setExpanded(newExpanded ? panel : false)
  }

  return (
    <ExpansionPanel square expanded={expanded === 'panel1'} onChange={handleChange('panel1')}>
      <ExpansionPanelSummary aria-controls='panel1d-content' id='panel1d-header'>
        <Typography>1ο Εξάμηνο</Typography>
      </ExpansionPanelSummary>
      <ExpansionPanelDetails style={{ padding: 0 }}>
        <Table className={classes.table}>
          <TableHead style={{ border: 0 }}>
            <TableRow>
              <StyledTableCell>Τίτλος</StyledTableCell>
              <StyledTableCell align='center'>department</StyledTableCell>
              <StyledTableCell align='center'>semester</StyledTableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {rows.map(row => (
              <StyledTableRow key={row.name}>
                <StyledTableCell component='th' scope='row' allign='left'>
                  {row.name}
                </StyledTableCell>
                <StyledTableCell align='center'>{row.department.name}</StyledTableCell>
                <StyledTableCell align='center'>{row.semester}</StyledTableCell>
              </StyledTableRow>
            ))}
          </TableBody>
        </Table>
      </ExpansionPanelDetails>
    </ExpansionPanel>
  )
}
