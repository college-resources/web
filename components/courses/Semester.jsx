import { useState } from 'react'
import { makeStyles, withStyles } from '@material-ui/core/styles'
import Accordion from '@material-ui/core/Accordion'
import AccordionDetails from '@material-ui/core/AccordionDetails'
import AccordionSummary from '@material-ui/core/AccordionSummary'
import ExpandMoreIcon from '@material-ui/icons/ExpandMore'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import Typography from '@material-ui/core/Typography'
import { lightBlue } from '@material-ui/core/colors'

const StyledTableCell = withStyles((theme) => ({
  head: {
    backgroundColor: theme.palette.secondary.main,
    color: theme.palette.common.white
  },
  root: {
    borderBottom: 0
  }
}))(TableCell)

const StyledTableRow = withStyles((theme) => ({
  root: {
    '&:nth-of-type(even)': {
      backgroundColor:
        theme.palette.type === 'light'
          ? lightBlue[50]
          : theme.palette.primary.light
    },
    '&:nth-of-type(odd)': {
      backgroundColor:
        theme.palette.type === 'light'
          ? lightBlue[100]
          : theme.palette.primary.level3
    }
  }
}))(TableRow)

const useStyles = makeStyles((theme) => ({
  icon: {
    color: theme.palette.common.white
  },
  root: {
    width: '100%'
  },
  summary: {
    backgroundColor: theme.palette.primary.dark,
    color: theme.palette.common.white
  }
}))

export default function Semester(props) {
  const classes = useStyles()
  const { semester } = props
  const { rows } = props
  const [expanded, setExpanded] = useState(false)

  const handleChange = (panel) => (event, newExpanded) => {
    setExpanded(newExpanded ? panel : false)
  }

  return (
    <div className={classes.root}>
      <Accordion
        className={classes.summary}
        expanded={expanded === `panel${semester}`}
        onChange={handleChange(`panel${semester}`)}
      >
        <AccordionSummary
          aria-controls="panel-content"
          expandIcon={<ExpandMoreIcon className={classes.icon} />}
          id="panel1d-header"
        >
          <Typography>
            Εξάμηνο&nbsp;
            {semester}
          </Typography>
        </AccordionSummary>
        <AccordionDetails
          style={{ overflowX: 'auto', overflowY: 'hidden', padding: 0 }}
        >
          <Table>
            <TableHead>
              <TableRow>
                <StyledTableCell
                  align="center"
                  rowSpan="2"
                  style={{ minWidth: '16px', width: '32px' }}
                >
                  Κωδ.
                </StyledTableCell>
                <StyledTableCell rowSpan="2" style={{ minWidth: '10rem' }}>
                  Τίτλος
                </StyledTableCell>
                <StyledTableCell
                  align="center"
                  rowSpan="2"
                  style={{ minWidth: '16px', width: '32px' }}
                >
                  Είδος
                </StyledTableCell>
                <StyledTableCell
                  align="center"
                  colSpan="2"
                  style={{
                    borderBottom: '1px solid rgba(224, 224, 224, 1)',
                    minWidth: '16px',
                    width: '32px'
                  }}
                >
                  Ώρες
                </StyledTableCell>
                <StyledTableCell
                  align="center"
                  rowSpan="2"
                  style={{ minWidth: '16px', width: '32px' }}
                >
                  Π.Μ.
                </StyledTableCell>
              </TableRow>
              <TableRow>
                <StyledTableCell align="center">Θ</StyledTableCell>
                <StyledTableCell align="center">Ε</StyledTableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {rows.map((row) => (
                <StyledTableRow key={row._id}>
                  <StyledTableCell align="center">
                    {row.lessonCode}
                  </StyledTableCell>
                  <StyledTableCell align="left">{row.name}</StyledTableCell>
                  <StyledTableCell align="center">{row.type}</StyledTableCell>
                  <StyledTableCell align="center">
                    {row.hoursTheory}
                  </StyledTableCell>
                  <StyledTableCell align="center">
                    {row.hoursLab}
                  </StyledTableCell>
                  <StyledTableCell align="center">{row.credit}</StyledTableCell>
                </StyledTableRow>
              ))}
            </TableBody>
          </Table>
        </AccordionDetails>
      </Accordion>
    </div>
  )
}
