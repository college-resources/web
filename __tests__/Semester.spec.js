import Semester from '../components/Semester'
import renderer from 'react-test-renderer'

describe('App', () => {
  const data = [
    {
      _id: 1,
      lessonCode: 1,
      name: 'lesson1',
      semester: 1,
      type: 'E',
      hoursTheory: 4,
      hoursLab: 2,
      credit: 6
    },
    {
      _id: 2,
      lessonCode: 2,
      name: 'lesson2',
      semester: 1,
      type: 'E',
      hoursTheory: 4,
      hoursLab: 2,
      credit: 6
    }
  ]

  it('renders', () => {
    const component = renderer.create(<Semester rows={data} semester={1} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  // it('fetches async data', () => {
  //   const promise = new Promise((resolve, reject) =>
  //     setTimeout(
  //       () =>
  //         resolve({
  //           data: {
  //             lessons: [
  //               {
  //                 _id: 1,
  //                 lessonCode: 1,
  //                 name: 'lesson1',
  //                 semester: 1,
  //                 type: 'E',
  //                 hoursTheory: 4,
  //                 hoursLab: 2,
  //                 credit: 6
  //               }
  //             ]
  //           }
  //         }),
  //       100
  //     )
  //   )
  //
  //   promise.then(() => {
  //     const wrapper = mount(<Semester rows={this.data} semester={1} />)
  //
  //     expect(wrapper.find('tbody').find('tr').length).toEqual(2)
  //   })
  // })
})
