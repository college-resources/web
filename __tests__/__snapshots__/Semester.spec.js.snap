// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root MuiExpansionPanel-root makeStyles-summary-3 MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-controls="panel-content"
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root"
      id="panel1d-header"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Εξάμηνο 
          1
        </p>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root makeStyles-icon-2"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
